include/maliput/routing/route.h

```diff
@@ -0,0 +1,122 @@
+//
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+
+#include <optional>
+#include <vector>
+
+#include "maliput/api/lane_data.h"
+#include "maliput/api/regions.h"
+#include "maliput/api/road_network.h"
+#include "maliput/common/maliput_copyable.h"
+#include "maliput/common/maliput_throw.h"
+#include "maliput/routing/route_phase.h"
+
+namespace maliput {
+namespace routing {
+
+/// Describes the sequence of paths to go from one RoadPosition to another.
+/// It is sequenced by RoutePhases which contain default and adjacent
+/// maliput::api::LaneSRanges which conduce the agent from the start to the end
+/// road position.
+///
+/// Agents are expected to use the Router to obtain a Route. Once in the Route,
+/// they can iterate through the RoutePhases or find by a specific INERTIAL or
+/// LANE Frame coordinate the RoutePhase and start driving from there towards
+/// the end goal.
+///
+/// The first RoutePhase start road position identifies the beginning of the
+/// Route. The last RoutePhase end road position identifies the ending of the
+/// Route. RoutePhases in the sequence are connected end to end by the same
+/// INERTIAL-Frame coordinate whose the corresponding LANE-Frame coordinate
+/// depends on which maliput::api::LaneSRange they contain.
+class Route final {
+ public:
+  MALIPUT_DEFAULT_COPY_AND_MOVE_AND_ASSIGN(Route);
+  Route() = delete;
+
+  /// Constructs a Route.
+  ///
+  /// @param route_phases The sequence of RoutePhases. It must not be nullptr.
+  /// RoutePhases must be connected end to end.
+  /// @param road_network The maliput::api::RoadNetwork pointer. It must not be
+  /// nullptr.
```

Comment:
I see that an alias to this pointer is maintained as a member variable. Thus, should document the lifetime requirements of this parameter. Something like: "The lifetime of this pointer must exceed that of this object."

Reply:
Done.

Reply:
No change?

Reply:
I think it is solved now.